File: CS/Sample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList;

namespace Sample
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            InitData();
            label3.Text = "This example demonstrates how to use DragDrop capabilities of the XtraTreeList within the control and when working with another control.";

            treeList1.OptionsBehavior.AutoChangeParent = false;
            treeList1.OptionsBehavior.CloseEditorOnLostFocus = false;
            treeList1.OptionsBehavior.DragNodes = true;
            treeList1.OptionsBehavior.ShowToolTips = false;
            treeList1.OptionsBehavior.SmartMouseHover = false;
            treeList1.OptionsSelection.MultiSelect = true;

            listBox1.AllowDrop = true;
            listBox1.ItemHeight = 16;
        }

        private void InitData()
        {
            DevExpress.XtraTreeList.Design.XViews xv = new DevExpress.XtraTreeList.Design.XViews(treeList1);
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void listBox1_DragEnter(object sender, DragEventArgs e)
        {
            if ( GetDragNodes(e.Data) != null)
                e.Effect = DragDropEffects.Copy;
        }

        private void listBox1_DragDrop(object sender, DragEventArgs e)
        {
            DevExpress.XtraEditors.ListBoxControl lb = sender as DevExpress.XtraEditors.ListBoxControl;
            DevExpress.XtraTreeList.TreeListMultiSelection nodes = GetDragNodes(e.Data);
            if (nodes != null)
            {
                int ind = lb.IndexFromPoint(lb.PointToClient(new Point(e.X, e.Y)));                
                for (int i = 0 ; i < nodes.Count ; i++)
                {
                    string dragString  = GetStringByNode(nodes[i]);
                    if (ind == -1)
                        lb.Items.Add(dragString);
                    else
                        lb.Items.Insert(ind, dragString);                    
                }
            }
        }

        private DevExpress.XtraTreeList.TreeListMultiSelection GetDragNodes(IDataObject data)
        {
             return data.GetData(typeof (DevExpress.XtraTreeList.TreeListMultiSelection)) as  DevExpress.XtraTreeList.TreeListMultiSelection;
        }

        private string GetStringByNode(DevExpress.XtraTreeList.Nodes.TreeListNode node)
        {
            string ret  = string.Empty;
            for (int i = 0 ; i < treeList1.Columns.Count; i++)
                ret += node.GetDisplayText(i) + (i < treeList1.Columns.Count - 1 ? "; ": ".");
            return ret;
        }
        private TreeListHitInfo dragStartHitInfo; 
        private void treeList1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left && Control.ModifierKeys == Keys.None)
            {
                DevExpress.XtraTreeList.TreeList tl = sender as DevExpress.XtraTreeList.TreeList;
                dragStartHitInfo = tl.CalcHitInfo(e.Location);
            }
        }

        private void treeList1_MouseMove(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left && dragStartHitInfo != null && dragStartHitInfo.Node != null){
                Size dragSize = SystemInformation.DragSize;
                Rectangle dragRect = new Rectangle(new Point(dragStartHitInfo.MousePoint.X - dragSize.Width / 2,
                    dragStartHitInfo.MousePoint.Y - dragSize.Height / 2), dragSize);
                if ( !dragRect.Contains(e.Location) )
                    ((TreeList)sender).DoDragDrop(treeList1.Selection, DragDropEffects.Copy);
            }

        }

        private void splitter1_SplitterMoved(object sender, SplitterEventArgs e)
        {

        }
    }
}

File: VB/Sample/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList

Namespace Sample
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			InitData()
			label3.Text = "This example demonstrates how to use DragDrop capabilities of the XtraTreeList within the control and when working with another control."

			treeList1.OptionsBehavior.AutoChangeParent = False
			treeList1.OptionsBehavior.CloseEditorOnLostFocus = False
			treeList1.OptionsBehavior.DragNodes = True
			treeList1.OptionsBehavior.ShowToolTips = False
			treeList1.OptionsBehavior.SmartMouseHover = False
			treeList1.OptionsSelection.MultiSelect = True

			listBox1.AllowDrop = True
			listBox1.ItemHeight = 16
		End Sub

		Private Sub InitData()
			Dim xv As New DevExpress.XtraTreeList.Design.XViews(treeList1)
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

		End Sub

		Private Sub listBox1_DragEnter(ByVal sender As Object, ByVal e As DragEventArgs) Handles listBox1.DragEnter
			If GetDragNodes(e.Data) IsNot Nothing Then
				e.Effect = DragDropEffects.Copy
			End If
		End Sub

		Private Sub listBox1_DragDrop(ByVal sender As Object, ByVal e As DragEventArgs) Handles listBox1.DragDrop
			Dim lb As DevExpress.XtraEditors.ListBoxControl = TryCast(sender, DevExpress.XtraEditors.ListBoxControl)
			Dim nodes As DevExpress.XtraTreeList.TreeListMultiSelection = GetDragNodes(e.Data)
			If nodes IsNot Nothing Then
				Dim ind As Integer = lb.IndexFromPoint(lb.PointToClient(New Point(e.X, e.Y)))
				For i As Integer = 0 To nodes.Count - 1
					Dim dragString As String = GetStringByNode(nodes(i))
					If ind = -1 Then
						lb.Items.Add(dragString)
					Else
						lb.Items.Insert(ind, dragString)
					End If
				Next i
			End If
		End Sub

		Private Function GetDragNodes(ByVal data As IDataObject) As DevExpress.XtraTreeList.TreeListMultiSelection
			 Return TryCast(data.GetData(GetType(DevExpress.XtraTreeList.TreeListMultiSelection)), DevExpress.XtraTreeList.TreeListMultiSelection)
		End Function

		Private Function GetStringByNode(ByVal node As DevExpress.XtraTreeList.Nodes.TreeListNode) As String
			Dim ret As String = String.Empty
			For i As Integer = 0 To treeList1.Columns.Count - 1
				If i < treeList1.Columns.Count - 1 Then
					ret &= node.GetDisplayText(i) + ("; ")
				Else
					ret &= node.GetDisplayText(i) + (".")
				End If
			Next i
			Return ret
		End Function
		Private dragStartHitInfo As TreeListHitInfo
		Private Sub treeList1_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs) Handles treeList1.MouseDown
			If e.Button = MouseButtons.Left AndAlso Control.ModifierKeys = Keys.None Then
				Dim tl As DevExpress.XtraTreeList.TreeList = TryCast(sender, DevExpress.XtraTreeList.TreeList)
				dragStartHitInfo = tl.CalcHitInfo(e.Location)
			End If
		End Sub

		Private Sub treeList1_MouseMove(ByVal sender As Object, ByVal e As MouseEventArgs) Handles treeList1.MouseMove
			If e.Button = MouseButtons.Left AndAlso dragStartHitInfo IsNot Nothing AndAlso dragStartHitInfo.Node IsNot Nothing Then
				Dim dragSize As Size = SystemInformation.DragSize
				Dim dragRect As New Rectangle(New Point(dragStartHitInfo.MousePoint.X - dragSize.Width \ 2, dragStartHitInfo.MousePoint.Y - dragSize.Height \ 2), dragSize)
				If (Not dragRect.Contains(e.Location)) Then
					CType(sender, TreeList).DoDragDrop(treeList1.Selection, DragDropEffects.Copy)
				End If
			End If

		End Sub

		Private Sub splitter1_SplitterMoved(ByVal sender As Object, ByVal e As SplitterEventArgs)

		End Sub
	End Class
End Namespace